src/osemgrep/cli_ci/Ci_CLI.ml

```diff
@@ -125,23 +108,19 @@ let cmdline_term caps : conf Term.t =
    * it below so we can get a nice man page documenting those environment
    * variables (Romain's idea).
    *)
-  let combine scan_conf audit_on beta_testing_secrets code dry_run
-      _internal_ci_scan_results secrets supply_chain suppress_errors _git_meta
-      _github_meta _historical_secrets =
+  let combine scan_conf audit_on code dry_run _internal_ci_scan_results
+      supply_chain suppress_errors _git_meta _github_meta _historical_secrets =
     let products =
-      (if beta_testing_secrets || secrets then [ `Secrets ] else [])
-      @ (if code then [ `SAST ] else [])
-      @ if supply_chain then [ `SCA ] else []
+      (if code then [ `SAST ] else []) @ if supply_chain then [ `SCA ] else []
```

Comment:
I'm not sure how to populate the `products` var here. I had to end up getting rid of `o_beta_testing_secrets` since we're dropping that flag, and `o_secrets` because it complained about this flag already existing in `scan_cli.ml`

Also I'm a little confused why the secrets flags `o_no_secrets_validation` and `o_allow_untrusted_validators` exists in `Scan_CLI.ml` but `o_historical_secrets` exists in `Ci_CLI.ml`. Could it be that historical scanning only makes sense for CI?

cc @kopecs

Reply:
> I'm not sure how to populate the products var here with secrets. I had to end up getting rid of o_beta_testing_secrets since we're dropping that flag, and o_secrets because it complained about this flag already existing in scan_cli.ml

You should be able to move `o_secrets` to here; the `--secrets` flag only makes sense for CI, yes?

I believe previously we had `--beta-testing-secrets-enabled` to serve the purpose of telling the CLI that the product was available. However, I think we may now be getting that from the App?

> Also I'm a little confused why the secrets flags o_no_secrets_validation and o_allow_untrusted_validators exists in Scan_CLI.ml but o_historical_secrets exists in Ci_CLI.ml. Could it be that historical scanning only makes sense for CI?

I believe I did this on your PR adding it since it was just added to `ci.py`. If we want to make it a common option (some group in `scan.py` plus `cli_scan/Common_CLI.ml` I think) I think that's reasonable.

Reply:
Just wanted to confirm that someone **won't** have to pass the `--secrets` flag if the app config has Secrets enabled

Reply:
Yes, I agree. Aiui the status quo generally is:
- `semgrep scan` runs all products you have (based on info from app) with the given config.
- For `scan`, `--config <product>` can be used to run only the policy associated with a given product (e.g., only code rules, only secrets rules, only supply chain); multiple `--config`s could be passed if desired.
- `semgrep ci` runs the policy you have defined in app
- For `ci`, you can pass `--<product>` to run only that product instead of what is enabled via the app. Multiple such flags can be passed.

So I would think the `--secrets` flag falls into that CI bucket and we don't need it for `scan`.

Reply:
@kopecs that makes sense to me and I tend to agree. I also have to remove `--secrets` from scan then. @leifdreizler do we know if we have any customers or sales folks relying on `semgrep scan --secrets`?